tests/stubtest_third_party.py

```diff
@@ -34,6 +34,11 @@ def run_stubtest(dist: Path, *, verbose: bool = False) -> bool:
         print(colored("skipping", "yellow"))
         return True
 
+    platform = stubtest_meta.get("platform", None)
+    if platform is not None and sys.platform not in platform:
```

Comment:
Wouldn't it make more sense just to have this field default to "only test on linux" for now?

```suggestion
    platforms_to_test = stubtest_meta.get("platform", ["linux"])
    if sys.platform not in platforms_to_test:
```

Reply:
No, because we want more coverage for more packages. Most of the packages do not rely on platform, so we can be quite safe.

Reply:
I agree that increasing coverage in the long term is an important goal. However, the consensus in #8660 was that it isn't necessary for most stubs to run stubtest on multiple platforms, so it would just slow down CI.

I'd prefer it if this PR focused on adding the necessary infrastructure, without changing the default for all stubs. Then, after this PR is merged, we can experiment with changing the default for all stubs, and evaluate whether it significantly slows down CI (or leads to widespread CI failures on stubs that haven't been stubtested on Windows before).